test/dynamo/test_generator.py

```diff
@@ -993,6 +993,252 @@ def fn(t):
         self.assertEqual(z, 2)
 
 
+class TestGeneratorThrow(GeneratorTestsBase):
+    def test_throw(self):
+        def whoo(t):
+            try:
+                yield t.sin()
+            except RuntimeError:
+                yield t.cos()
+
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            gen = whoo(t)
+            a = next(gen)
+            b = gen.throw(RuntimeError)
+            return a + b
+
+        t = torch.randn(2)
+        y = fn(t)
+        self.assertEqual(y, t.sin() + t.cos())
+
+    def test_throw_with_finally(self):
+        z = 0
+
+        def whoo():
+            nonlocal z
+            z = 0
+            try:
+                try:
+                    yield 1
+                except ValueError:
+                    yield 2
+                finally:
+                    z += 2
+            except ValueError:
+                z += 3
+                yield 4
+            finally:
+                z += 1
+            z += 10
+
+        @torch.compile(fullgraph=True, backend="eager")
+        def f(x):
+            gen = whoo()
+            next(gen)
+            gen.throw(ValueError)
+            return x.sin()
+
+        f(torch.randn(3))
+        self.assertEqual(z, 3)
```

Comment:
You should compare f(torch.randn(3)) with f(torch.randn(3)) without torch.compile to check consistency. Sometimes cpython surprises me with its behavior.